### PR TITLE
Prepare for integrations and add Infoblox integration

### LINKED ISSUE #128: Migrate Infoblox SSoT Plugin to This Repo
### Environment
* Nautobot version:  
* nautobot-plugin-chatops version:  

### Proposed Functionality

Following https://github.com/nautobot/nautobot-plugin-chatops/issues/194, and establishing #115, merge https://github.com/nautobot/nautobot-plugin-ssot-infoblox into this repo as an `integration`.

### Use Case

Make installation, discovery, maintaining easier.

### PR DESCRIPTION
**Pull request by snaselj**
# Closes #128 #115 

- Adds Infoblox integration from nautobot-plugin-ssot-infoblox.
- Adds dynamic integration loading.

## TODO

- [x] Test integration.

### CODE REVIEW

**Comment by snaselj** [comment]
This PR is ready for review now. Integration is not tested as I do not have access to any Infoblox instance. Commit https://github.com/nautobot/nautobot-plugin-ssot/pull/134/commits/82b006adb2a273b9942855088a1566f7eb602091 contains changes to files copied from nautobot-plugin-ssot-infoblox.

File: nautobot_ssot/models.py
Comment (Kircheneer) [inline nautobot_ssot/models.py:208]: Can we maybe solve this using some kind of registration pattern that allows us not to mess with `globals`? It looks like a little too much magic for my taste 😃 

If not, I would at least appreciate a docstring and some comments here.

File: nautobot_ssot/urls.py
Comment (Kircheneer) [inline nautobot_ssot/urls.py:25]:
```suggestion
        urlpatterns.extend(module.urlpatterns)
```

File: nautobot_ssot/utils.py
Comment (Kircheneer) [inline nautobot_ssot/utils.py:4]: I don't quite understand why this change is necessary?

Also:
```suggestion
"""Utility functions for Nautobot SSoT App."""
import logging

logger = logging.getLogger("nautobot.ssot")
```

File: pyproject.toml
Comment (Kircheneer) [inline pyproject.toml:63]:
```suggestion
```

**Review by Kircheneer — CHANGES_REQUESTED** [review]
I did not review the Infoblox-related code as I assume it comes from an established plugin?

File: nautobot_ssot/models.py
Comment (snaselj) [reply nautobot_ssot/models.py:208]: Other option is to use `sys.modules`, that needs to be imported. Using `globals()` seems better for me, as it doesn't need to be imported. Will add comments.

File: docs/dev/dev_environment.md
Comment (snaselj) [reply docs/dev/dev_environment.md:365]: This was probably copied from ChatOps plugin without changing the text, I just fixed it.

File: nautobot_ssot/integrations/infoblox/urls.py
Comment (jdrew82) [inline nautobot_ssot/integrations/infoblox/urls.py:1]: If we're not doing any specific links in the UI we should probably remove this file.

**Review by jdrew82 — CHANGES_REQUESTED** [review]
Mostly looks good. Just a few questions and things to address before I can approve though.

File: docs/dev/dev_environment.md
Comment (snaselj) [reply docs/dev/dev_environment.md:365]: Related to `Slack`, it's not an Integration, it's Chat Provider. There is no similar entity in SSoT App, that's why I changed `Slack` -> `Nautobot`. Any suggestions welcome.

File: development/nautobot_config.py
Comment (snaselj) [reply development/nautobot_config.py:15]: Several reasons :-)

- `is_truthy` is using `strtobool`, that's deprecated in Python 3.10, will be removed.
- `get_bool_env` does more that `is_truthy` (reads from env, prefixing `NAUTOBOT_SSOT` here), IMHO easier to implement locally using standard libs over custom implementation.

File: development/nautobot_config.py
Comment (snaselj) [reply development/nautobot_config.py:150]: Same as above

File: nautobot_ssot/__init__.py
Comment (snaselj) [reply nautobot_ssot/__init__.py:12]: I prefer less text, seems to me clear. Opinions?

File: nautobot_ssot/integrations/infoblox/urls.py
Comment (snaselj) [reply nautobot_ssot/integrations/infoblox/urls.py]: Removed

File: development/docker-compose.base.yml
Comment (snaselj) [reply development/docker-compose.base.yml]: Updated as proposed

File: development/nautobot_config.py
Comment (snaselj) [reply development/nautobot_config.py:15]: I can rename these functions to `_get_ssot_env` or `_get_ssot_bool_env`, would that be more clear?

**Comment by snaselj** [comment]
Tested and passed `Infoblox ⟹ Nautobot` job import.

File: development/nautobot_config.py
Comment (jdrew82) [reply development/nautobot_config.py:15]: I wasn't aware of the deprecation. If that's the case, it might make more sense to update that function in the core app to your method as I guarantee there are a lot of configurations in use out there using `is_truthy` that's going to break in the future. Better to fix that and continue using it instead of creating some replacement.

Comment (snaselj) [reply development/nautobot_config.py]: Fixed as proposed.

File: nautobot_ssot/__init__.py
Comment (snaselj) [reply nautobot_ssot/__init__.py]: Fixed

**Review by jdrew82 — CHANGES_REQUESTED** [review]
Still a misunderstanding of the change requested.

File: development/nautobot_config.py
Comment (jdrew82) [reply development/nautobot_config.py:15]: I'm unclear as to why using `is_truthy()` would be less readable as it's been our standard pattern for a while. If we want to switch to these newer methods to be more concise on what they're doing that's fine. I just didn't understand why we were replicating existing functionality that's widely used already, even in the core app's dev environment.

Comment (snaselj) [reply development/nautobot_config.py:15]: To answer your question:

With local functions:

```
        "hide_example_jobs": _get_ssot_bool_env("HIDE_EXAMPLE_JOBS"),
        "infoblox_default_status": _get_ssot_env("INFOBLOX_DEFAULT_STATUS", "active"),
        "infoblox_enable_rfc1918_network_containers": _get_ssot_bool_env("INFOBLOX_ENABLE_RFC1918_NETWORK_CONTAINERS"),
        "infoblox_enable_sync_to_infoblox": _get_ssot_bool_env("INFOBLOX_ENABLE_SYNC_TO_INFOBLOX"),
        "infoblox_import_objects_ip_addresses": _get_ssot_bool_env("INFOBLOX_IMPORT_OBJECTS_IP_ADDRESSES"),
        "infoblox_import_objects_subnets": _get_ssot_bool_env("INFOBLOX_IMPORT_OBJECTS_SUBNETS"),
        "infoblox_import_objects_vlan_views": _get_ssot_bool_env("INFOBLOX_IMPORT_OBJECTS_VLAN_VIEWS"),
        "infoblox_import_objects_vlans": _get_ssot_bool_env("INFOBLOX_IMPORT_OBJECTS_VLANS"),
        "infoblox_import_subnets": _get_ssot_env("INFOBLOX_IMPORT_SUBNETS").split(","),
        "infoblox_password": _get_ssot_env("INFOBLOX_PASSWORD"),
        "infoblox_url": _get_ssot_env("INFOBLOX_URL"),
        "infoblox_username": _get_ssot_env("INFOBLOX_USERNAME"),
        "infoblox_verify_ssl": _get_ssot_bool_env("INFOBLOX_VERIFY_SSL", True),
        "infoblox_wapi_version": _get_ssot_env("INFOBLOX_WAPI_VERSION", "v2.12"),
```

Without local functions:

```
        "infoblox_default_status": os.getenv("NAUTOBOT_SSOT_INFOBLOX_DEFAULT_STATUS", "active"),
        "infoblox_enable_rfc1918_network_containers": is_truthy(
            os.getenv("NAUTOBOT_SSOT_INFOBLOX_ENABLE_RFC1918_NETWORK_CONTAINERS")
        ),
        "infoblox_enable_sync_to_infoblox": is_truthy(os.getenv("NAUTOBOT_SSOT_INFOBLOX_ENABLE_SYNC_TO_INFOBLOX")),
        "infoblox_import_objects_ip_addresses": is_truthy(
            os.getenv("NAUTOBOT_SSOT_INFOBLOX_IMPORT_OBJECTS_IP_ADDRESSES")
        ),
        "infoblox_import_objects_subnets": is_truthy(os.getenv("NAUTOBOT_SSOT_INFOBLOX_IMPORT_OBJECTS_SUBNETS")),
        "infoblox_import_objects_vlan_views": is_truthy(os.getenv("NAUTOBOT_SSOT_INFOBLOX_IMPORT_OBJECTS_VLAN_VIEWS")),
        "infoblox_import_objects_vlans": is_truthy(os.getenv("NAUTOBOT_SSOT_INFOBLOX_IMPORT_OBJECTS_VLANS")),
        "infoblox_import_subnets": os.getenv("NAUTOBOT_SSOT_INFOBLOX_IMPORT_SUBNETS").split(","),
        "infoblox_password": os.getenv("NAUTOBOT_SSOT_INFOBLOX_PASSWORD"),
        "infoblox_url": os.getenv("NAUTOBOT_SSOT_INFOBLOX_URL"),
        "infoblox_username": os.getenv("NAUTOBOT_SSOT_INFOBLOX_USERNAME"),
        "infoblox_verify_ssl": is_truthy(os.getenv("NAUTOBOT_SSOT_INFOBLOX_VERIFY_SSL", True)),
        "infoblox_wapi_version": os.getenv("NAUTOBOT_SSOT_INFOBLOX_WAPI_VERSION", "v2.12"),
```

My personal opinion is that later is less readable. :-)

Decision is up to you, should I completely remove those local functions?

Comment (snaselj) [reply development/nautobot_config.py]: Changed as requested, thanks @bryanculver to unblock this.

**Comment by snaselj** [comment]
Resolved conflicts with develop.

File: development/nautobot_config.py
Comment (jdrew82) [reply development/nautobot_config.py]: Thanks for updating those. Looks good to me now!

**Review by jdrew82 — APPROVED** [review]
LGTM!

**Pull request merged**
